communication/include/communication/CommDefs.hpp

```diff
@@ -108,6 +110,9 @@ struct TlsTcpConfig : PlainTcpConfig {
   // https://www.openssl.org/docs/man1.0.2/man1/ciphers.html
   std::string cipherSuite;
 
+  bool usePKDecryption;
```

Comment:
Maybe we can use std::optional for SecretData. I don't insist though - it's up to you.